app/models/host/managed.rb

```diff
@@ -922,9 +922,10 @@ def trigger_nic_orchestration
   # For performance reasons logs and reports are deleted in batch
   # see http://projects.theforeman.org/issues/8316 for details
   def remove_reports
-    return if reports.empty?
-    Log.where("report_id IN (#{reports.pluck(:id).join(',')})").delete_all
-    Report.where("host_id = #{id}").delete_all
+    host_reports = Report.where("host_id = #{id}")
+    return if host_reports.count == 0
+    Log.where("report_id IN (#{host_reports.pluck(:id).join(',')})").delete_all
```

Comment:
you can use `Log.where(report_id: host_reports.pluck(:id))` here instead of creating the sql yourself. Then you don't have to check the count first, it will just generate two delete statements that will do nothing if there are no matching reports. (It would be better to actually use a nested query but IIRC mysql doesn't support nested queries for delete)

Reply:
I just blindly copied the code but I'm happy to refactor as part of this :-) I think the `count==0` still makes sense to avoid running 3 queries with `1=0` - pluck and two deletes.

Reply:
I must say i disagree :) running count when there are many records can take some time; running 3 queries with a `1=0` condition takes 0 time - the database knows there's nothing to do.

Reply:
updated as you wish, I guess you're right